cmd/snap-confine/mount-support.c

```diff
@@ -62,47 +62,82 @@
 // TODO: fold this into bootstrap
 static void setup_private_mount(const char *snap_name)
 {
-	char tmpdir[MAX_BUF] = { 0 };
+	// TODO: remove after switching to global umask handling.
+	mode_t old_mask = umask(0);
 
-	// Create a 0700 base directory, this is the base dir that is
-	// protected from other users.
+	// Create a 0700 base directory, this is the base dir that is protected
+	// from other users. This directory name is NOT randomly generated. This
+	// has several properties:
```

Comment:
This comment is incomplete. You both create a 0700 basedir and a 1777 tmpdir. Also, on every invocation you unconditionally chmod and chown them (which is fine in terms of security).